### PR TITLE
Add support for JPA Embedded/Embeddable annotations

### LINKED ISSUE #552: Add support for JPA Embedded/Embeddable annotations
JPA `Entity` annotation is already supported. Please also add support for `Embedded`/`Embeddable` JPA annotations with the semantics similar to what  is described in JPA specification.

### PR DESCRIPTION
**Pull request by andrey-vasilyev**

### CODE REVIEW

**Comment by Kontakter** [comment]
Could you please name this Pull Request more informative? I think you could use the title from the issue

**Comment by andrey-vasilyev** [comment]
> Could you please name this Pull Request more informative? I think you could use the title from the issue

done, thanks

File: yt/java/ytsaurus-client/src/test/java/tech/ytsaurus/client/rows/SchemaOfEmbeddedEntityTest.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+import javax.persistence.Embeddable;
+import javax.persistence.Embedded;
+import javax.persistence.Entity;
+import java.time.ZonedDateTime;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+public class SchemaOfEmbeddedEntityTest {
```
Comment (tinarsky) [inline yt/java/ytsaurus-client/src/test/java/tech/ytsaurus/client/rows/SchemaOfEmbeddedEntityTest.java:17]: You should write tests for the following cases:
1. the class has a nested entity (not embedded), and it has an embedded field;
2. in the class there is an embedded field, in which there is a field with a decimal, without specifying the precision and scale in the annotation (in this case, you need to pass the table schema with the specified precision and scale to the create method);
3. integration tests for reading/writing embedded classes (currently only schema construction is implemented, please add the code to EntitySkiffSerializer).

**Review by tinarsky — CHANGES_REQUESTED** [review]

File: yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/rows/EntityTableSchemaCreator.java [hunk 2]
```diff
@@ -36,22 +38,49 @@ private EntityTableSchemaCreator() {
...
+                                                     StructType tableSchemaAsStructType,
+                                                     Map<Class<?>, String> fieldsChain) {
+        if (fieldsChain.containsKey(clazz)) {
+            throw new InfiniteLoopException();
+        }
+        for (Field field : getAllDeclaredFields(clazz)) {
             if (isFieldTransient(field, JavaPersistenceApi.transientAnnotations())) {
                 continue;
             }
+            if (anyOfAnnotationsPresent(field.getType(), JavaPersistenceApi.embeddableAnnotations())) {
```
[the remark this answers is not in the document]
Comment (andrey-vasilyev) [reply yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/rows/EntityTableSchemaCreator.java:71]: Such behavior seem to inline with JPA spec and Oracle tutorials. Here is SO [topic](https://stackoverflow.com/questions/31157700/is-the-jpa-embedded-annotation-mandatory) and Oracle [tutorial](https://docs.oracle.com/cd/E19798-01/821-1841/gjiwz/index.html), which clearly says: 

> Entities that own embeddable classes as part of their persistent state may annotate the field or property with the javax.persistence.Embedded annotation but are not required to do so.

File: yt/java/ytsaurus-client/src/test/java/tech/ytsaurus/client/rows/SchemaOfEmbeddedEntityTest.java (hunk 1 above)
Comment (andrey-vasilyev) [reply yt/java/ytsaurus-client/src/test/java/tech/ytsaurus/client/rows/SchemaOfEmbeddedEntityTest.java:17]: 1. Isn't this  case already  covered in `SchemaOfEntityTest`
2 and 3 will do, thanks

**Comment by Kontakter** [comment]
Let rebase this PR and will try to finish it

**Comment by andrey-vasilyev** [comment]
> Let rebase this PR and will try to finish it

sorry, urgent matters at work, will try to finish it this week

**Comment by andrey-vasilyev** [comment]
Hello @tinarsky could you please check again and provide feedback? Thank you.

**Comment by tinarsky** [comment]
Hello, I'll check it in the next few days.

**Review by tinarsky — CHANGES_REQUESTED** [review]

File: yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/rows/EntityFieldDescr.java [hunk 3]
```diff
@@ -27,13 +29,27 @@ boolean isTransient() {
...
+
     public List<Type> getTypeParameters() {
         return typeParameters;
     }
 
     static List<EntityFieldDescr> of(List<Field> fields) {
         return fields.stream()
                 .map(EntityFieldDescr::new)
-                .collect(Collectors.toUnmodifiableList());
+                .toList();
```
[the remark this answers is not in the document]
Comment (andrey-vasilyev) [reply yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/rows/EntityFieldDescr.java]: fixed

**Review by tinarsky — APPROVED** [review]
LGTM

**Pull request closed without merging**